src/Adapter/Swoole/EventLoop.php

```diff
@@ -0,0 +1,298 @@
+<?php
+
+namespace M6Web\Tornado\Adapter\Swoole;
+
+use Generator;
+use M6Web\Tornado\Adapter\Swoole\Internal\DummyPromise;
+use M6Web\Tornado\Deferred;
+use M6Web\Tornado\Promise;
+use Swoole\Coroutine;
+use RuntimeException;
+use Swoole\IDEHelper\StubGenerators\Swoole;
+use Swoole\Runtime;
+use Throwable;
+use function extension_loaded;
+
+class EventLoop implements \M6Web\Tornado\EventLoop
+{
+    private $cids;
+
+    public function __construct()
+    {
+        if (!extension_loaded('swoole')) {
+            throw new RuntimeException(
+                'EventLoop must running only with swoole extension.'
+            );
+        }
+
+        $this->cids = [];
+    }
+
+    private function shiftCoroutine(): mixed
+    {
+        if(count($this->cids) === 0) {
+            return null;
+        }
+
+        $cid = array_shift($this->cids);
+        if(Coroutine::exists($cid)) {
+            Coroutine::resume($cid);
```

Comment:
But we should resume a coroutine only if the `DummyPromise` (blocking related generator) is resolved… why to always resume each coroutine?

Reply:
Actually this is the case as all promises for `Swoole\EventLoop` are instanciated as below

```
return new DummyPromise(function () {
  $this->shiftCoroutine();
});
```

At resolution time of the `DummyPromise`, this will resume the first Swoole Coroutine in the queue.